providers/stores/azblob.js

```diff
@@ -33,11 +34,13 @@ class AzBlobStore extends AbstractStore {
           this.containerName,
           name,
           continuation,
+          { include: azure.BlobUtilities.BlobListingDetails.METADATA },
           resultOrError(resolve, reject)
         )
       })
       result.entries.forEach(entry => {
-        const value = this._getEntry(entry.name, type)
+        const value =
+          type === 'entity' ? this._getEntry(entry.metadata.coordinates, type) : this._getEntry(entry.name, type)
```

Comment:
Just to be clear, only the Entities (i.e., definitions) will have `coordinates` in the blob metadata? Could/should we just put it in there all the time?

Note the new code in PR adds a new type of thing being stored in blobs. It currently adds a new instance of azblob using a different container (sort of how we do it for definitions) to store the interesting file content. These could be classified as `results` since they really do go with the actual tool results, we've just chosen to store them in separate files.

Reply:
I figured we should put it on all blobs and stop referring to blob names whenever we want coordinates. The following is my current plan + understanding, please correct me if I am on the wrong track:

---

when you call `list(coordinates, 'entity')` you are looking for something that looks like

```
[ 'npm/npmjs/-/co/4.6.0' ]
```

this will respect casing

---

when you call `list(coordinates, 'result')` you are looking for something that looks like

```
[ 'npm/npmjs/-/co/4.6.0/clearlydefined/1, 'npm/npmjs/-/co/4.6.0/scancode/2.2.1' ]
```

this will always be lowercased, driven from the blob name itself

---

we will put the metadata on the blobs that live here:

harvested:
 - `npm/npmjs/-/co/revision/4.6.0/tool/clearlydefined/1.json`
 - `npm/npmjs/-/co/revision/4.6.0/tool/scancode/2.2.1.json`

cached:
 - `npm/npmjs/-/co/revision/4.6.0/tool/definition/1.json`

and that metadata will be `coordinates=npm/npmjs/-/co/4.6.0`

Reply:
Hmmm, I think we should be consistent. If the result coords are lowered then the caller cannot reasonably/reliably convert back and forth. If we just say that coords "case preserving" then it is simple and easy for people to understand. 

Regarding putting the metadata on all blobs, I agree but it seems like the code has two different behaviors by triggering off the `type` passed to `list`. If the canonical coords were stored in the metadata and all coords were case preserving then the code and user story would be simplified.

Reply:
If I call the list function with type=result do I want to get back blob names?

If we preserve the casing of the coordinate, it no longer will point to a blob

Reply:
you want to get back a coordinates object. if it is cased, that's ok as we'll go back to the store at some point and ask for the content related to the coordinates. The store implementation will (or perhaps not) toLowerCase the coordinate elements to form the name/path/id to look for in the store. 

Casing in the store infrastructure is a topic for the store. The inputs and outputs are coordinates. It would be much clearer/stronger if we could say that they are case preserving.

Does that work?

Reply:
Yep makes sense :+1: